### ISSUE TITLE
[BUG] 'NoneType' object has no attribute 'span'

### ISSUE THREAD

**Issue by Impre-visible** [pytube/pytube#1242]
**Before creating an issue**

Please confirm that you are on the latest version of pytube by installing from the source.
You can do this by running `python -m pip install git+https://github.com/pytube/pytube`.
Sometimes, the pypi library repository is not up to date, and your issue may have been fixed already!
im at the latest version
**Describe the bug**
i want to download a video, but i get this error when i do this : 

**To Reproduce**
Please provide the following information:
- The video or playlist url that is causing the error.
- The code where the problem is occurring.
```py
video.streams.first().download()
```
that simple code
**Expected behavior**
i just want to download the video

**Output**
If pytube raises an exception, please provide the full traceback for the exception.
```py
Traceback (most recent call last):
  File "G:\AlbumDownloader\app.py", line 13, in <module>
    start()
  File "G:\AlbumDownloader\app.py", line 11, in start
    searchSong(Playlist(url), artiste)
  File "G:\AlbumDownloader\base.py", line 13, in searchSong
    video.streams.first().download()
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\__main__.py", line 292, in streams
    return StreamQuery(self.fmt_streams)
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\__main__.py", line 177, in fmt_streams
    extract.apply_signature(stream_manifest, self.vid_info, self.js)
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\extract.py", line 409, in apply_signature
    cipher = Cipher(js=js)
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\cipher.py", line 43, in __init__
    self.throttling_plan = get_throttling_plan(js)
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\cipher.py", line 387, in get_throttling_plan
    raw_code = get_throttling_function_code(js)
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\cipher.py", line 301, in get_throttling_function_code
    code_lines_list = find_object_from_startpoint(js, match.span()[1]).split('\n')
AttributeError: 'NoneType' object has no attribute 'span'
```
**System information**
Please provide the following information:
- Python version : 3.10.0
- Pytube version 11.0.2
- Command used to install pytube pip and python -m pip install git+https://github.com/pytube/pytube

**Comment by er-nabin-bhusal** [comment]
Having same issue here.

**Comment by iamDyeus** [comment]
yeah even am having the same issue.

**Comment by taehjoon** [comment]
I found it here
https://stackoverflow.com/questions/70060263/pytube-attributeerror-nonetype-object-has-no-attribute-span

in ciper.py on line 293
change 
`name = re.escape(get_throttling_function_name(js))`
to
`name="hha"`

and this problem already mentioned 
https://github.com/pytube/pytube/issues/1233#issue-1123428834

**Comment by Impre-visible** [comment]
not worked, again an error 
```py
Traceback (most recent call last):
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\__main__.py", line 177, in fmt_streams
    extract.apply_signature(stream_manifest, self.vid_info, self.js)
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\extract.py", line 409, in apply_signature
    cipher = Cipher(js=js)
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\cipher.py", line 29, in __init__
    self.transform_plan: List[str] = get_transform_plan(js)
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\cipher.py", line 197, in get_transform_plan
    return regex_search(pattern, js, group=1).split(";")
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\helpers.py", line 129, in regex_search
    raise RegexMatchError(caller="regex_search", pattern=pattern)
pytube.exceptions.RegexMatchError: regex_search: could not find match for re.escape(get_initial_function_name(js))=function\(\w\){[a-z=\.\(\"\)]*;(.*);(?:.+)}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "G:\AlbumDownloader\app.py", line 13, in <module>
    start()
  File "G:\AlbumDownloader\app.py", line 11, in start
    searchSong(Playlist(url), artiste)
  File "G:\AlbumDownloader\base.py", line 13, in searchSong
    video.streams.first().download()
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\__main__.py", line 292, in streams
    return StreamQuery(self.fmt_streams)
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\__main__.py", line 184, in fmt_streams
    extract.apply_signature(stream_manifest, self.vid_info, self.js)
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\extract.py", line 409, in apply_signature
    cipher = Cipher(js=js)
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\cipher.py", line 29, in __init__
    self.transform_plan: List[str] = get_transform_plan(js)
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\cipher.py", line 197, in get_transform_plan
    return regex_search(pattern, js, group=1).split(";")
  File "C:\Users\imprv\AppData\Local\Programs\Python\Python310\lib\site-packages\pytube\helpers.py", line 129, in regex_search
    raise RegexMatchError(caller="regex_search", pattern=pattern)
pytube.exceptions.RegexMatchError: regex_search: could not find match for re.escape(get_initial_function_name(js))=function\(\w\){[a-z=\.\(\"\)]*;(.*);(?:.+)}
```

**Comment by Impre-visible** [comment]
hoo, i found the problem, thanks

**Comment by glubsy** [comment]
Duplicate of #1218, fix available in #1222.
